### PR TITLE
changing post_auth_check for post_key_auth

### PR DESCRIPTION
**Pull request by carlosvillanua**
post_key_auth is the parameter that needs to be used in the tyk API classic schema not post_auth_check

## Types of changes

- [x] Fixing typo (please merge to production)
- [ ] Documenting a new feature (please merge to production)
- [ ] Documentation for future release (please do not merge to production)
- [ ] Something else (please add if needs merging to production or not)

## Checklist

- [x] I have [reviewed the guidelines](../CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](../CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [ ] I have added a preview link.

### CODE REVIEW

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 1]
```diff
@@ -224,7 +224,7 @@ All types of custom middleware hooks are supported by Tyk Golang plugins. They r
 2.`"use_go_plugin_auth": true`
 
 #### Post Authentication Hook
-`"post_auth_check"` hook can only be use when:
+`"post_key_auth"` hook can only be use when:
```
Comment (letzya) [inline tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
`"post_key_auth"` hook can only be used under the following circumstances:
```

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 2]
```diff
@@ -214,7 +214,7 @@ All types of custom middleware hooks are supported by Tyk Golang plugins. They r
 
 * `"pre"` - contains array of middlewares to be run before any others (i.e. before authentication).
 * `"auth_check"` - contains only one middleware info, his middleware performs custom authentication and adds API key session info into request context.
-* `"post_auth_check"` - contains array of middlewares to be run after authentication, at this point we have authenticated session API key for the given key (in request context) so we can perform any extra checks.
+* `"post_key_auth"` - contains array of middlewares to be run after authentication, at this point we have authenticated session API key for the given key (in request context) so we can perform any extra checks.
```
Comment (dcs3spp) [inline tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
* `"post_key_auth"` - contains array of middlewares to be run after authentication. At this point we have authenticated session API key for the given key (in request context) so we can perform any extra checks.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
lgtm with minor corrections

**Pull request merged**